mex/backend/extracted/models.py

```diff
@@ -1,29 +1,23 @@
 from enum import Enum, EnumMeta, _EnumDict
-from typing import TYPE_CHECKING, Generator, Union, cast
+from typing import TYPE_CHECKING, Iterator, Literal, Union, cast
 
-from pydantic import create_model
+from pydantic import Field, create_model
 
-from mex.backend.models import TypeSerializer
 from mex.common.models import BASE_MODEL_CLASSES, BaseExtractedData, BaseModel
 from mex.common.transform import dromedary_to_snake
-from mex.common.types import Identifier, PrimarySourceID
 
 
 def _collect_extracted_model_classes(
     base_models: list[type[BaseModel]],
-) -> Generator[tuple[str, type[BaseExtractedData]], None, None]:
+) -> Iterator[tuple[str, type[BaseExtractedData]]]:
```

Comment:
Copied from the [mypy cheatsheet](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#functions). I find it easier to remember and to understand since we do not use the extra arguments of Generators. I'm also open for reverting back to Generator. What do you think?

Reply:
i just liked the old guideline of **Return the most specific type, accept the most generic**. 

i'd be open to change generators to iterator for the sake of YAGNI, but then i'd ask you to change it in the entire mex codebase. so pick your poison ;)

Reply:
alright, I give in: ea914361f12f7136ff13cf7e7e90a43746fa10b8